lib/pub_sub/subscriber.rb

```diff
@@ -5,55 +5,65 @@ def self.subscribe
     end
 
     def subscribe
-      topics = PubSub.config.subscriptions.keys.map do |service_identifier|
-        subscribe_to_service(service_identifier)
+      PubSub.config.regions.each do |region|
+        topics = PubSub.config.subscriptions.keys.map do |service_identifier|
+          subscribe_to_service(service_identifier, region)
+        end
+        set_queue_policy(topics.map(&:arn), region)
       end
-      set_queue_policy(topics.map(&:arn))
     end
 
     private
 
-    def subscribe_to_service(service_identifier)
-      topic = Aws::SNS::Topic.new(sns.create_topic(name: service_identifier).topic_arn)
-      topic.subscribe(endpoint: queue_arn, protocol: 'sqs')
-      PubSub.logger.info "Subscribed #{queue_arn} to #{topic.arn}"
+    def subscribe_to_service(service_identifier, region)
+      topic = Aws::SNS::Topic.new(sns(region).create_topic(name: service_identifier).topic_arn, region: region)
+      topic.subscribe(endpoint: queue_arn(region), protocol: 'sqs')
+      PubSub.logger.info "Subscribed #{queue_arn(region)} to #{topic.arn}"
       topic
     end
 
-    def set_queue_policy(topic_arns)
-      Aws::SQS::Client.new.set_queue_attributes(
-        queue_url: queue_url,
+    def set_queue_policy(topic_arns, region)
+      Aws::SQS::Client.new(region: region).set_queue_attributes(
+        queue_url: queue_url(region),
         attributes: {
           Policy: JSON.generate(
-            policy_attributes(topic_arns)
+            policy_attributes(topic_arns, region)
           )
         }
       )
     end
 
-    delegate :queue_url, :queue_arn, to: :queue
-
-    def policy_attributes(topic_arns)
+    def policy_attributes(topic_arns, region)
       {
         Version: '2008-10-17',
-        Id: "#{queue_arn}/SQSDefaultPolicy",
+        Id: "#{queue_arn(region)}/SQSDefaultPolicy",
         Statement: [{
-          Sid: "SendMessageTo##{queue_arn}",
+          Sid: "SendMessageTo##{queue_arn(region)}",
           Effect: 'Allow',
           Principal: { 'AWS': '*' },
           Action: 'SQS:SendMessage',
-          Resource: queue_arn,
+          Resource: queue_arn(region),
           Condition: { ArnEquals: { 'aws:SourceArn': topic_arns } }
         }]
       }
     end
 
-    def sns
-      @sns ||= Aws::SNS::Client.new
+    def sns(region)
+      Aws::SNS::Client.new(region: region)
```

Comment:
Shouldn't subscriber, publisher, and queue - all use the same `AWS Client` constructor? It might make more sense to put it under PubSub directly and have local sns and sqs methods of these 3 call that constructor